active/0002-centalized-mgmt-resource-types-provider-versions.md

```diff
@@ -0,0 +1,262 @@
+
+## Motivation
+
+The [resource type providers dynamic loading feature](https://github.com/Azure/bicep/issues/5453) introduces a change in provider declaration statement syntax to include the registry address and version that Bicep will use to resolve resource type declarations.
+
+**Provider Declaration Syntax**
+
+```bicep
+provider 'br:mcr.microsoft.com/bicep/providers/az@1.2.3' // fully-qualified
+provider 'br/public:az@1.2.3' // uses provider aliasing defined in bicepconfig.json
+```
+
+We also currently support the following syntax for "built-in" providers:
+
+```bicep
+provider 'sys@1.0.0'
+```
+
+In the above cases, the provider can be referenced by using its symbol which is derived from the segment to the left of the '@' sign. (`az` and `sys` for the declarations above)
+
+Both syntaxes also support aliasing of that symbol reference:
+
+```bicep
+provider 'br/public:az@0.2.3' as myAz
+provider 'sys@1.0.0' as mySys
+```
+
+The current syntax (shown above) presents several challenges:
+
+1. The new syntax is quite verbose, and cannot be controlled centrally. For users with a large set of files, picking up the latest version of `az` will require modifying every single file.
+1. The aliasing (assuming that the last segment of the registry namespace == the provider name) & versioning (having an `@` character instead of a `:` which is familiar to module users) behavior of the new syntax can be confusing.
+1. Its deceiving and confusing to allow `sys` to have a pinned versionis fixed version `1.0.0` - this **should** imply that there are no changes between Bicep versions, which isn't realistic.
+
+- It's an "either/or" - either users follow the current behavior (`az` imported by default) **or** they're using the new syntax. We don't have a plan for how to combine the two.
+- With [Radius](https://github.com/radius-project/radius), it's conceivable that we may want users to **not** have `az` imported by default.
+
+### Proposed Changes
+
+- [Updating `bicepconfig.json` merge semantics](#updating-bicepconfigjson-merge-semantics)
+- [Resolution of namespace identifiers by VSCode extension when inspecting published module sources](#resolution-of-namespace-identifiers-by-vscode-extension-when-inspecting-published-module-sources)
+- [Wildcard semantics for provider versions in `bicepconfig.json`](#wildcard-semantics-for-provider-versions-in-bicepconfigjson)
+
+## Detailed design
+
+### Client side changes
+
+#### Add a new syntax for the provider import declaration statements: `provider {providerName} [as {optionalAlias}]`
+
+The example below shows a usage of the proposed syntax
+
+```bicep
+provider az
+provider kubernetes as k8s // uses optional aliasing of the
+```
+
+The resolution of the symbols will be resolved from the builtin `bicepconfig.json` under a new section:
+
+```json
+{
+  // prior bicepconfig.json sections
+  // ...
+
+  "providers": {
+    "az": {
```

Comment:
are the dictionary keys going to be validated? In other words, if `az` does not exist at the location `mcr.microsoft.com/bicep/providers` will an error be thrown?

Reply:
During the compilation process if the artifact reference `mcr.microsoft.com/bicep/providers/az:0.2.3` cannot be resolved for any reason, the system will emit a diagnostic with a reasonable error message that informs the user regarding the nature of the failure if one is faound.